client/signup/steps/domains/index.tsx

```diff
@@ -307,7 +308,7 @@ const DomainSearchUI = (
 		const [ sitesBackLabelText, defaultBackUrl ] =
 			userSiteCount && userSiteCount === 1
 				? [ __( 'Back to My Home' ), '/home' ]
-				: [ __( 'Back to sites' ), hostingDashboardOptIn ? '/v2/sites' : '/sites' ];
+				: [ __( 'Back to sites' ), hostingDashboardOptIn ? dashboardLink( '/sites' ) : '/sites' ];
```

Comment:
This might not applicable anymore but it was changed in https://github.com/Automattic/wp-calypso/pull/106804 with the other opt-in path update logic.